src/data/mv_polynomial/rename.lean

```diff
@@ -105,6 +105,25 @@ begin
   exact finsupp.map_domain_injective (finsupp.map_domain_injective hf)
 end
 
+section
+variable (s : set σ)
+open_locale classical
+
+/-- Given a subset `s` of the set of variables `σ`, `kill_compl s` is the `alg_hom`
+  from `R[σ]` to `R[s]` that sends the variables in the complement of `s` to `0`. -/
+def kill_compl : mv_polynomial σ R →ₐ[R] mv_polynomial s R :=
```

Comment:
I would prefer to avoid an `_`. How about `retract`? And can you please generalize to `σ ↪ τ`, instead of only set inclusions?

Reply:
Since retracts aren't unique I think the name should be indicative of what the retract does: here I choose the retract that sends the extra variables to 0, hence the name. It's the same as the existing `mv_polynomial.const_coeff` when the larger polynomial ring is viewed as a polynomial ring over the smaller polynomial ring, so in some sense preferred, but you are free to send the extra variables to everything.

Generalizing to embeddings will break some defeq's, but maybe is something we should do ... I'll look into it.